libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -50,13 +50,12 @@ void StaticAnalyzer::endVisit(ContractDefinition const&)
 
 bool StaticAnalyzer::visit(FunctionDefinition const& _function)
 {
-	const bool isInterface = m_currentContract->contractKind() == ContractDefinition::ContractKind::Interface;
 	if (_function.noVisibilitySpecified())
 		m_errorReporter.warning(
 			_function.location(),
 			"No visibility specified. Defaulting to \"" +
-			(isInterface ? "external" : Declaration::visibilityToString(_function.visibility())) +
-			"\"."
+			Declaration::visibilityToString(_function.visibility()) +
+			"\". In interfaces it defaults to external."
```

Comment:
Perhaps to avoid confusion in some cases, this could be conditionally displayed on `isInterface`.